salt/modules/nilrt_ip.py

```diff
@@ -284,6 +284,57 @@ def _get_requestmode_info(interface):
     return 'dhcp_linklocal'
 
 
+def _get_adaptermode_info(interface):
+    '''
+    return adaptermode for given interface
+    '''
+    with salt.utils.files.fopen(INI_FILE, 'r') as config_file:
+        config_parser = configparser.RawConfigParser(dict_type=CaseInsensitiveDict)
+        config_parser.read_file(config_file)
+        mode = '' if not config_parser.has_option(interface, 'mode') else \
+            _remove_quotes(config_parser.get(interface, 'mode').lower())
```

Comment:
This seems to be an anti-pattern - doing an 'has_option' then a 'get()'. Can we use a default for the get() and do only that?

More than that, the first 5 lines seems to be a pattern that is repeated throughout the code. Can/Should we create a higher-level wrapper over the configparser to hide all this boilerplate code?

Reply:
`get()` throws an exception if the option doesn't exist 

We can make a class or a global variable where the config will be read only one time, but the problem is that if we write something with `nirtcfg` then we should reload config from **INI_FILE** in configparser, so a method like this one will be ok:
```python
def _get_option_configparser(section, option, default_value='', file=INI_FILE):
        with salt.utils.files.fopen(file, 'r') as config_file:
            config_parser = configparser.RawConfigParser(dict_type=CaseInsensitiveDict)
            config_parser.read_file(config_file)
            if config_parser.has_section(section) and config_parser.has_option(option):
                return config_parser.get(section, option)
            return default_value
```

Reply:
There should be some option to pass defaults to the get, so you can avoid the has_option/get pair.

We shouldn't need a class, but only a function that can take a list of keys and returns a dict with keys/values